File: entity/src/main/java/far/galaxy/foodcourt/entity/group/CustomerGroup.java
package far.galaxy.foodcourt.entity.group;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import far.galaxy.foodcourt.entity.customer.Customer;
import org.springframework.lang.NonNull;

import javax.persistence.*;
import java.util.List;

@Entity
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class CustomerGroup {
    private long id;
    private String title;

    private List<Customer> customers;

    public CustomerGroup() {
        this.id = 0;
    }

    public CustomerGroup(String title) {
        this();
        this.title = title;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    @NonNull
    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    @OneToMany(cascade=CascadeType.ALL, fetch = FetchType.LAZY)
    public List<Customer> getCustomers() {
        return customers;
    }

    public void setCustomers(List<Customer> customers) {
        this.customers = customers;
    }
}

File: white/src/main/javascript/far/galaxy/foodcourt/web/entity/CustomerGroup.js
import Customer from "./Customer";

export default class CustomerGroup {

    constructor(value) {
        this.value = value;
    }

    getId() {
        return this.value.id;
    }

    getTitle() {
        return this.value.title;
    }

    getCustomers() {
        if (this.value.customers && this.value.customers.length) {
            return this.value.customers.map(customer => new Customer(customer));
        }
        return null;
    }
}
